publication-rest/src/main/java/no/unit/nva/publication/delete/DeletePublicationHandler.java

```diff
@@ -62,44 +65,74 @@ protected Void processInput(Void input, RequestInfo requestInfo, Context context
         var publicationIdentifier = RequestUtil.getIdentifier(requestInfo);
 
         var publication = resourceService.getPublicationByIdentifier(publicationIdentifier);
-        var publicationStatus = publication.getStatus();
-
-        switch (publicationStatus) {
-            case PUBLISHED:
-                if (!PublicationPermissionStrategy.fromRequestInfo(requestInfo).hasPermissionToUnpublish(publication)) {
-                    throw new UnauthorizedException();
-                }
-                var duplicate = requestInfo.getQueryParameterOpt(DUPLICATE_QUERY_PARAM).orElse(null);
-                resourceService.unpublishPublication(toPublicationWithDuplicate(duplicate, publication));
-                persistNotification(publication);
-                break;
-            case DRAFT:
-                resourceService.markPublicationForDeletion(userInstance, publicationIdentifier);
-                break;
-            default:
-                throw new BadRequestException(
-                    String.format("Publication status %s is not supported for deletion", publicationStatus));
+
+        switch (publication.getStatus()) {
+            case DRAFT -> handleDraftDeletion(userInstance, publicationIdentifier);
+            case PUBLISHED -> handleSoftDeletion(requestInfo, publication);
+            case UNPUBLISHED -> handleHardDeletion(requestInfo, publication);
+            default -> unsupportedPublicationForDeletion(publication);
         }
 
         return null;
     }
 
-    private void persistNotification(Publication publication) throws ApiGatewayException {
-        TicketEntry.requestNewTicket(publication, UnpublishRequest.class).persistNewTicket(ticketService);
-    }
-
     @Override
     protected Integer getSuccessStatusCode(Void input, Void output) {
         return HttpStatus.SC_ACCEPTED;
     }
 
+    private static void validateHardDeletionRequest(RequestInfo requestInfo, Publication publication)
+        throws UnauthorizedException {
+        if (isEditor(requestInfo, publication) || canEditOwnInstitutionResources(requestInfo)) {
```

Comment:
A Curator shouldn't be allowed to hard delete a publication, if I recall correctly?